### PR TITLE
(DOCSP-34756): C++: Update code examples for GA

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

Jira ticket: https://jira.mongodb.org/browse/DOCSP-34756

- [C++ SDK Staged Changes](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-34756/sdk/cpp/)

This PR updates the code examples for the C++ GA. 

- Removes the experimental namespace
- Removes the alpha and beta test suites
- Removes the Current/Deprecated tabs on the docs pages
- Removes the `Tip` directive about features using the experimental namespace
- Removes "beta" naming from all code examples

This does not remove any beta/preview language from the docs themselves, or change the code examples except minor tweaks required to get the tests to pass.

For review, I'm looking for you to check for:

- Code examples that should be on the page are present
- I've appropriately Bluehawk replaced realm object names to simpler versions
- I've removed the Current/Deprecated tabs on all pages
- I haven't accidentally removed any text that was in the Current tab of a Current/Deprecated tab
- Any model names referred to in the copy on the page matches the model names in the code examples

I've done a preliminary check, fixed some things, and I think it looks good, but I'm sure I've missed things.

Separate PRs will follow to update the SDK naming, remove version directives, update API reference links, and add new functionality.

A few code examples are currently commented out but will still show up on the pages. Support for BSON has been removed from the SDK. I'll split out separate tickets to address those code examples. 

I am not looking for grammar or wording changes for any page or code example content, unless I've introduced an erroneous change in this PR. The scope of changes is too large for me to address grammar/content errors here. If you spot something egregious, feel free to start a document or Jira ticket with things for me to address as fast follows.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/cpp/app-services/call-a-function.txt [hunk 1]
```diff
@@ -44,7 +44,7 @@ which we provide as a ``BsonArray`` of arguments:
 .. versionchanged:: 0.2.0 Returns std::future instead of std::promise
```
Comment (cbullinger) [inline source/sdk/cpp/app-services/call-a-function.txt:44]: Just confirming: per your PR description, these version directives will all be removed as a separate body of work, yeah?

File: source/sdk/cpp/sync/stream-data-to-atlas.txt [hunk 2]
```diff
@@ -26,35 +26,15 @@ Sync Data Unidirectionally from a Client Application
...
-
-            .. literalinclude:: /examples/generated/cpp/asymmetric-sync.snippet.asymmetric-object.cpp
-               :language: cpp
+      .. literalinclude:: /examples/generated/cpp/asymmetric-sync.snippet.asymmetric-object.cpp
+         :language: cpp
 
-            For more information on how to define an ``asymmetric_object``, including
-            limitations when linking to other object types, see: 
-            :ref:`Define an Asymmetric Object <cpp-define-asymmetric-object>`.
+      For more information on how to define an ``REALM_ASYMMETRIC_SCHEMA``,
```
Comment (cbullinger) [inline source/sdk/cpp/sync/stream-data-to-atlas.txt:35]:
```suggestion
      For more information on how to define a ``REALM_ASYMMETRIC_SCHEMA``, 
```

File: source/sdk/cpp/sync/stream-data-to-atlas.txt [hunk 3]
```diff
@@ -65,42 +45,22 @@ Sync Data Unidirectionally from a Client Application
...
-            using a ``flexibleSyncConfiguration()``. Unlike opening other 
-            types of realms in the ``experimental`` namespace, when you 
-            open a realm for Data Ingest, you *must* specify the
-            ``asymmetric_object`` types you want to sync.
-
-            .. literalinclude:: /examples/generated/cpp/asymmetric-sync.snippet.beta-open-asymmetric-synced-realm.cpp
-               :language: cpp
+      After you have an authenticated user, you can open a synced realm
+      using a ``flexibleSyncConfiguration()``. Unlike opening other 
+      types of realms in the ``experimental`` namespace, when you
```
Comment (cbullinger) [inline source/sdk/cpp/sync/stream-data-to-atlas.txt:55]: should this reference to the ``experimental`` namespace also be removed?

File: source/sdk/cpp/sync/stream-data-to-atlas.txt [hunk 4]
```diff
@@ -117,22 +77,9 @@ Sync Data Unidirectionally from a Client Application
...
-               :language: cpp
-
-            .. include:: /includes/tip-cpp-beta-experimental-features.rst
-
-         .. tab:: Deprecated
-            :tabid: deprecated
-
-            .. literalinclude:: /examples/generated/cpp/asymmetric-sync.snippet.create-asymmetric-object.cpp
-               :language: cpp
+      .. literalinclude:: /examples/generated/cpp/asymmetric-sync.snippet.create-asymmetric-object.cpp
```
Comment (cbullinger) [inline source/sdk/cpp/sync/stream-data-to-atlas.txt:80]: is the indentation on this example correct?

File: examples/cpp/asymmetric/asymmetric-sync.cpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  auto realm = realm::open<realm::WeatherSensorReading>(syncConfig);
+  // :snippet-end:
+
+  // Doing this for simplicity in testing, but should not be relevant to
+  // dev code examples, so hiding it.
+  const realm::object_id oid = realm::object_id::generate();
+
+  // :snippet-start: create-asymmetric-object
+  auto weatherSensorReading =
+      realm::WeatherSensorReading{._id = oid,  // :remove:
```
Comment (cbullinger) [inline examples/cpp/asymmetric/asymmetric-sync.cpp:37]: looks like you need to update the indentation on the other properties too, they're _really_ indented
```suggestion
      realm::WeatherSensorReading{
      ._id = oid,  // :remove:
```

File: examples/cpp/asymmetric/asymmetric-sync.cpp [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+                                  .deviceId = "WX1278UIT",
+                                  .temperatureInFahrenheit = 64.7,
+                                  .windSpeedInMph = 7};
+
+  realm.write([&] { realm.add(std::move(weatherSensorReading)); });
```
Comment (cbullinger) [inline examples/cpp/asymmetric/asymmetric-sync.cpp:42]: was there a reason you reformatted this to one line? was written as multi-line previously: 

```
realm.write([&]{
   realm.add(std::move(weatherSensorReading));
});
```

File: source/sdk/cpp/sync/sync-subscriptions.txt [hunk 7]
```diff
@@ -37,33 +37,13 @@ To use Flexible Sync in your app, you must:
...
-      .. versionchanged:: v0.4.0-preview Replaces deprecated realm::App(...) with realm::App(const realm::App::configuration&)
-
-      .. literalinclude:: /examples/generated/cpp/flexible-sync.snippet.flexible-sync-prerequisites.cpp
-         :language: cpp
+.. literalinclude:: /examples/generated/cpp/flexible-sync.snippet.flexible-sync-prerequisites.cpp
+   :language: cpp
 
-      Additionally, models used with Sync must contain an identifier property 
-      named ``_id``.
+Additionally, when using the ``realm::experimental`` namespace, models
```
Comment (cbullinger) [inline source/sdk/cpp/sync/sync-subscriptions.txt:43]: same as above -- I assume this ref to ``realm::experimental`` should be removed?

File: examples/cpp/sync/authentication.cpp [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+  auto app = realm::App(appConfig);
+
+  auto user = app.login(realm::App::credentials::anonymous()).get();
+  // :snippet-end:
+  REQUIRE(!user.access_token().empty());
+  user.log_out().get();
+  REQUIRE(user.access_token().empty());
+}
+
+// TODO: Figure out how to do this properly in the updated SDK
```
Comment (cbullinger) [inline examples/cpp/sync/authentication.cpp:56]: was wondering why this entire snippet was commented out 😅

File: examples/cpp/sync/authentication.cpp [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+  appConfig.app_id = APP_ID;
+  auto app = realm::App(appConfig);
+
+  auto idToken = "<token>";
+
+  auto user = app.login(realm::App::credentials::apple(idToken)).get();
+  // :snippet-end:
+}
+
+// TODO: This has been renamed to realm::App::credentials::google_auth_code
```
Comment (cbullinger) [inline examples/cpp/sync/authentication.cpp:145]: aaand same for this and following test

**Review by cbullinger — COMMENTED** [review]
✅ pages in app-services, sync, and users

File: source/sdk/cpp/realm-files/configure-and-open-a-realm.txt [hunk 10]
```diff
@@ -23,39 +23,19 @@ Realm. A realm is a collection of the objects that you use
 in your application, called Realm objects, as well as additional metadata 
 that describe the objects.
 
-.. tabs::
+When opening a realm using the ``realm::experimental`` namespace,
```
Comment (cbullinger) [inline source/sdk/cpp/realm-files/configure-and-open-a-realm.txt:26]: noting `realm::experimental`

File: source/sdk/cpp/realm-files/configure-and-open-a-realm.txt [hunk 11]
```diff
@@ -137,28 +117,12 @@ Open the Default Realm
...
-   .. tab:: Current
-      :tabid: current
-
-      When opening a realm using the ``realm::experimental`` namespace, the 
-      C++ SDK Preview can automatically infer which models are available in the
-      project. You don't need to manually specify the available models.
-
-      .. literalinclude:: /examples/generated/cpp/crud.snippet.beta-open-realm.cpp
-         :language: cpp
+When opening a realm using the ``realm::experimental`` namespace, the
```
Comment (cbullinger) [inline source/sdk/cpp/realm-files/configure-and-open-a-realm.txt:120]: same

File: source/sdk/cpp/realm-files/configure-and-open-a-realm.txt [hunk 12]
```diff
@@ -289,29 +217,13 @@ Provide a Subset of Models to a Realm
...
-      of the ``realm::experimental::open()`` function.
+By default, the C++ SDK automatically adds all object types that have a
+:ref:`cpp-object-schema` in your executable to your 
+:ref:`cpp-realm-schema`. 
 
-      .. literalinclude:: /examples/generated/cpp/crud.snippet.beta-realm-specify-classes.cpp
-         :language: cpp
+However, if you want the realm to manage only a subset of models, you
+can specify those models by passing them into the template parameter list
+of the ``realm::experimental::open()`` function.
```
Comment (cbullinger) [inline source/sdk/cpp/realm-files/configure-and-open-a-realm.txt:226]: same

**Review by cbullinger — APPROVED** [review]
✅ remaining pages

Nice job! Only a few small things (and a handful that are prob out of scope for this PR -- feel free to ignore those).

File: source/sdk/cpp/app-services/call-a-function.txt (hunk 1 above)
Comment (dacharyc) [reply source/sdk/cpp/app-services/call-a-function.txt:44]: Yeah, I'll be removing the version directives when I work on this ticket: https://jira.mongodb.org/browse/DOCSP-34763

File: source/sdk/cpp/sync/stream-data-to-atlas.txt (hunk 4 above)
Comment (dacharyc) [reply source/sdk/cpp/sync/stream-data-to-atlas.txt:80]: Oh booger. The linter moved stuff around, so the Bluehawk :remove: line is now removing the entire first line of initializing the object. I guess I have to turn off the linter for this file? 🤔 

Or... I guess I have to use a replace for this, instead. Boo. Good catch, thank you!

File: source/examples/generated/cpp/crud.snippet.update-an-object.cpp [hunk 13]
```diff
@@ -1,17 +1,15 @@
 // Query for the object you want to update
-auto dogs = realm.objects<Dog>();
+auto dogs = realm.objects<realm::Dog>();
 // auto dogsNamedMaui = dogs.where("name == $0", {"Maui"});
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/examples/generated/cpp/crud.snippet.update-an-object.cpp:3]: Bah, yes, good catch!

File: examples/cpp/sync/authentication.cpp (hunk 8 above)
Comment (dacharyc) [reply examples/cpp/sync/authentication.cpp:56]: Yeah, I have a separate ticket for this and the following test. These were changed in a released version - NOT in this PR that removes the experimental namespace - so I'll fix these as part of the ticket I broke out: https://jira.mongodb.org/browse/DOCSP-35187

**Pull request merged**